src/peppr/match.py

```diff
@@ -166,105 +260,25 @@ def find_all_matches(
-    pose_entity_ids = entity_ids[len(reference_chains) :]
-    if (
-        np.bincount(reference_entity_ids, minlength=len(entity_ids))
-        != np.bincount(pose_entity_ids, minlength=len(entity_ids))
-    ).any():
-        raise UnmappableEntityError("Reference and pose have different entities")
-
-    anchor_index = _choose_anchor_chain(pose_chains, pose_entity_ids)
-
-    # Find corresponding chains by identifying the chain permutation minimizing the centroid RMSD
-    reference_centroids = np.array([struc.centroid(c) for c in reference_chains])
-    pose_centroids = np.array([struc.centroid(c) for c in pose_chains])
-    best_transform = None
-    best_rmsd = np.inf
-    best_chain_order = None
-    # Test all possible chains that represent the same entity against the anchor chain
-    for reference_i, reference_chain in enumerate(reference_chains):
-        if reference_entity_ids[reference_i] != pose_entity_ids[anchor_index]:
-            continue
-        else:
-            # Superimpose the entire system
-            # based on the anchor and chosen reference chain
-            transform = _get_superimposition_transform(
-                reference_chain, pose_chains[anchor_index]
-            )
-            chain_order = find_matching_centroids(
-                reference_centroids,
-                pose_centroids,
-                reference_entity_ids,
-                pose_entity_ids,
-                transform,
-            )
-            superimposed_pose_centroids = transform.apply(pose_centroids)
-            rmsd = struc.rmsd(
-                reference_centroids, superimposed_pose_centroids[chain_order]
-            )
-            if rmsd < best_rmsd:
-                best_rmsd = rmsd
-                best_transform = transform
-                best_chain_order = chain_order
-
-    pose_chains = [best_transform.apply(chain) for chain in pose_chains]  # type: ignore[union-attr]
-    return _create_indices_from_chain_order(
-        reference_chains,
-        pose_chains,
-        np.arange(len(reference_chains)),
-        best_chain_order,
-        # Superimposition is defined by centroids
-        superimpose=False,
-    )
-
-
 def find_matching_centroids(
     reference_centroids: NDArray[np.floating],
     pose_centroids: NDArray[np.floating],
     reference_entity_ids: NDArray[np.int_] | None = None,
     pose_entity_ids: NDArray[np.int_] | None = None,
-    transform: struc.AffineTransformation | None = None,
-) -> NDArray[np.int_]:
+) -> tuple[NDArray[np.int_], NDArray[np.int_]]:
     """
     Greedily find pairs of chains (each represented by its centroid) between the
     reference and the pose that are closest to each other.
 
     This functions iteratively chooses pairs with the smallest centroid distance, i.e.
     first the pair with the smallest centroid distance is chosen, then the pair with the
-    second smallest centroid distance and so on.
+    second smallest centroid distance and so on.∂
```

Comment:
not sure we need this :D 
```suggestion
    second smallest centroid distance and so on.
```